examples/materials/materialDatabase/src/index.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  createSession,
+  createViewport,
+  GeometryData,
+  IMaterialAbstractData,
+  IMaterialGemData,
+  IMaterialGemDataPropertiesDefinition,
+  IMaterialStandardData,
+  IMaterialStandardDataPropertiesDefinition,
+  ITreeNode,
+  MaterialEngine,
+  SessionOutputData,
+  VISIBILITY_MODE,
+} from '@shapediver/viewer';
+import { createUi } from '@shapediver/viewer.shared.demo-helper';
+
+/**
+* Create the material database of currently loaded materials.
+* 
+* The material database is a dictionary that maps the name of a material to its data.
+* The database is created by iterating over the material definitions provided by the MaterialDatabase output.
+*/
+const materialDatabase: { [key: string]: IMaterialStandardData | IMaterialGemData } = {};
+
+/**
+* Assign the materials from the material database to the geometries in the scene.
+* 
+* This function traverses the scene graph and assigns the materials from the material database to the geometries.
+* The material of a geometry is assigned based on the name of the material that is stored in the geometry data.
+* 
+* @param node 
+*/
+const assignMaterials = (node: ITreeNode) => {
+  node.traverse((c) => {
+      for (let i = 0; i < c.data.length; i++) {
+          const data = c.data[i];
+          if (data instanceof GeometryData) {
+              const materialName = data.material?.name;
+              if (materialName) {
+                  data.material = materialDatabase[materialName];
+                  data.updateVersion();
+                  node.updateVersion();
+              }
+          }
+      }
+  });
+};
+
+(async () => {
+  const viewport = await createViewport({
+      id: 'myViewport',
+      canvas: <HTMLCanvasElement>document.getElementById('canvas'),
+      // Set the visibility mode to manual to control the visibility of the scene manually.
+      visibility: VISIBILITY_MODE.MANUAL,
+  });
+
+  const session = await createSession({
+      id: 'mySession',
+      ticket: 'a4cb27a39d5abfc65036536db4bfead02fd9ee5e0b1f7d0630454fce5cfaa065fb8ab15821dbcb0136287534fdb6a1a5fd342c88e6948722336e6db5b1b8687c446e4741622c10d25476b6aed467b3afcf66c93e5e3d5070c8e850ff5bde5ee8093088e6936efe-59bd8515ca1f04d33b95507842220b8d',
+      modelViewUrl: 'https://sdr8euc1.eu-central-1.shapediver.com',
+  });
+
+  // Get the MaterialDatabase output.
+  const materialDatabaseOutput = session.getOutputByName('MaterialDatabase')[0];
+
+  /**
+   * Create a callback function that is called when the MaterialDatabase output is updated.
+   * In this callback, the materials are created from the material definitions and stored in the material database.
+   * 
+   * @param newNode 
+   * @returns 
+   */
+  const cb = (newNode?: ITreeNode) => {
```

Comment:
@MajorMeerkatThe3rd let's review the implementation of this function with regards to what we defined in this comment: https://shapediver.atlassian.net/browse/SS-7868?focusedCommentId=25950
I think what's missing is _Materials should be cached based on their name, type, and props, such that we avoid recreating materials._

Reply:
@snabela I'm not sure it's a good idea to cache anything. With lots of material changes we require more and more memory. Especially with materials that have many/large textures. I implemented it as specified now, but I don't think it's a good idea.

Reply:
@MajorMeerkatThe3rd ok I see what you mean. Let's discuss this briefly tomorrow in office. Many thanks!